packages/cli/src/x/actions/deploy.action.ts

```diff
@@ -73,6 +73,30 @@ async function handleAgentSelection(
   }
 }
 
+function findActiveTopics(allTopics: ListenerTopics[], client: any, topicsWithListeners = new Set()) {
+  client.listeners?.forEach((listener: ListenerTopics[]) => {
+    const listenerTopic = listener[0]
+    if (listenerTopic === '**') {
+      allTopics.forEach(topic => topicsWithListeners.add(topic))
+    } else if (listenerTopic.includes('**')) {
+      const [prefix] = listenerTopic.split(':')
+      allTopics.forEach(topic => { if (topic.split(':')[0] === prefix) topicsWithListeners.add(topic) })
+    } else if (allTopics.includes(listenerTopic)) {
+      topicsWithListeners.add(listenerTopic)
+    }
+  })
+  client.nodes?.forEach((nestedClient: any) => findActiveTopics(allTopics, nestedClient, topicsWithListeners))
+  return topicsWithListeners
+}
+
+async function getActiveTopics(file: string): Promise<Flatfile.EventTopic[]>{
+  const allTopics = Object.values(Flatfile.events.EventTopic)
+
+  const mount = await import(file);
+  return Array.from(findActiveTopics(allTopics, Client.create(mount.default))) as Flatfile.EventTopic[];
```

Comment:
Agree with Coderabbit that we should handle errors and fallbacks in case `mount.default` doesn't work. But we already don't do that so let's make that a separate enhancement and prioritize it!

Reply:
Yes, good cool, I'll put that up!